examples/lock-app/k32w/main/main.cpp

```diff
@@ -137,3 +187,151 @@ extern "C" void main_task(void const * argument)
+    {
+        SWITCH_DBG_LOG("woken up from FromKeyBoard");
+    }
+    else if (wakeReason.Bits.FromTMR == 1)
+    {
+        SWITCH_DBG_LOG("woken up from TMR");
+    }
+    dm_lp_wakeup();
+}
+
+static void dm_switch_preSleepCallBack(void)
+{
+    SWITCH_DBG_LOG("sleeping");
+
+    if (dualModeStates.threadInitialized)
+    {
+        otPlatRadioDisable(NULL);
+        dualModeStates.threadInitialized = FALSE;
+    }
+    /* Inform the low power dual mode module that we will sleep */
+    dm_lp_preSleep();
+
+    /* configure pins for power down mode */
+    BOARD_SetPinsForPowerMode();
+    /* DeInitialize application support for drivers */
+    BOARD_DeInitAdc();
+    /* DeInit the necessary clocks */
+    BOARD_SetClockForPowerMode();
+}
+
+void dm_switch_init15_4AfterWakeUp(void)
+{
+    uint32_t tick1 = 0;
+    uint32_t tick2 = 0;
+
+    if (dualModeStates.threadWarmBootInitTime == THREAD_WARM_BOOT_INIT_DURATION_DEFAULT_VALUE)
+    {
+        /* Get a 32K tick */
+        PWR_Start32kCounter();
+        tick1 = PWR_Get32kTimestamp();
+    }
+
+    /* Radio must be re-enabled after waking up from sleep.
+     * The module is completely disabled in power down mode */
+    ThreadExitSleep();
+
+    if (dualModeStates.threadWarmBootInitTime == THREAD_WARM_BOOT_INIT_DURATION_DEFAULT_VALUE)
+    {
+        tick2                                 = PWR_Get32kTimestamp();
+        dualModeStates.threadWarmBootInitTime = ((tick2 - tick1) * 15625u) >> 9;
+        /* Add a margin of 1 ms */
+        dualModeStates.threadWarmBootInitTime += 1000;
+    }
+}
+
+static void ThreadExitSleep()
+{
+    if (!dualModeStates.threadInitialized && ConnectivityMgr().IsThreadEnabled())
+    {
+        /* Install uMac interrupt handlers */
+        OSA_InstallIntHandler(ZIGBEE_MAC_IRQn, vMMAC_IntHandlerBbc);
+        OSA_InstallIntHandler(ZIGBEE_MODEM_IRQn, vMMAC_IntHandlerPhy);
+
+        /* Radio must be re-enabled after waking up from sleep.
+         * The module is completely disabled in power down mode */
+        otPlatRadioEnable(NULL);
+        dualModeStates.threadInitialized = TRUE;
+
+        /* wake up the Thread stack and check if any processing needs to be done */
+        otTaskletsSignalPending(NULL);
+    }
+}
+
+extern "C" void App_NotifyWakeup(void)
+{
+    if ((!dualModeStates.threadInitialized))
+    {
+        /* Notify the dual mode low power mode */
+        (void) App_PostCallbackMessage(dm_lp_processEvent, (void *) e15_4WakeUpEnded);
+    }
+}
+
+extern "C" void App_AllowDeviceToSleep()
+{
+    PWR_PreventEnterLowPower(false);
+}
+
+extern "C" void App_DisallowDeviceToSleep()
+{
+    PWR_PreventEnterLowPower(true);
+}
+
+/**
+ * Glue function called directly by the OpenThread stack
+ * when system event processing work is pending.
+ */
+extern "C" void otSysEventSignalPending(void)
+{
+    /* make sure that 15.4 radio is initialized before waking up the Thread task */
+    if (dualModeStates.threadInitialized == TRUE)
```

Comment:
This function is almost the same as the original version in `src/platform/K32W/ThreadStackManagerImpl.cpp`.

Why do we need this separate version with `if (dualModeStates.threadInitialized == TRUE)`?  Is that logic not valid in both builds?  If not, is there a way to organize this differently?  It's confusing that this function is in a different file based on whether we are in a low-power build.